HEN_HOUSE/egs++/sources/egs_dynamic_source/egs_dynamic_source.h

```diff
@@ -0,0 +1,277 @@
+P=P(i)+[P(i+1)-P(i)]/[mu(i+1)-mu(i)]*[R-mu(i)]
+Note that this scheme for generating incident source coordinates really
+only makes sense if mu(1)=0.0.  However, the source can function
+with mu(1)>0.0, in the case where a user desires to eliminate particles
+associated with a range of mu values, but there will be a lot of
+warning messages.
+ 
+A simple example is shown below.  This first defines a monoenergetic
+(1 MV) photon source in the Z-direction collimated to a 2x2 field
+centred on the Z-axis.  The control points place the source a
+distance, dsource, of 100 cm above the isocentre at (0,0,0).  Control
+points 1 and 2 rotate the source clockwise around the Y-axis (phi=0)
+through theta=0-360 degrees, while control points 3 and 4 rotate
+the source clockwise around the Z-axis (phi=90 degrees) through
+phi=0-360 degrees.  Note that mu(2)-mu(1)=mu(4)-mu(3), so the rotations
+around Z and Y are carried out for an equal number of incident photons.
+If the source being made to move dynamically supplies its own monitor
+unit index (iaea_phsp_source and egs_beam_source only), then the dynamic
+motion can be synchronized with the motion of component modules
+(MLC's, jaws) within the source by setting "synchronize motion"
+to "yes". 
+\verbatim
+:start source definition:
+    :start source:
+        library = egs_parallel_beam
+        name = my_parallel_source
+        :start shape:
+            library = egs_rectangle
+            rectangle = -.1 -.1 .1 .1
+        :stop shape:
+        direction = 0 0 1
+        charge = 0
+        :start spectrum:
+            type = monoenergetic
+            energy = 1.0
+        :stop spectrum:
+    :stop source:
+    :start source:
+        library = egs_dynamic_source
+        name = my_source
+        source name = my_parallel_source
+        :start motion:
+            control point 1 = 0 0 0 100 0 0 0 0
+            control point 2 = 0 0 0 100 360 0 0 0.5
+            control point 3 = 0 0 0 100 90 0 0 0.5 
+            control point 4 = 0 0 0 100 90 360 0 1.0 
+        :stop motion:
+    :stop source:
+
+    simulation source = my_source
+
+:stop source definition:
+\endverbatim
+*/
+
+class EGS_DYNAMIC_SOURCE_EXPORT EGS_DynamicSource :
+    public EGS_BaseSource {
+
+public:
+
+
+    struct EGS_ControlPoint {
+
+       EGS_Vector iso; //isocentre position
+       EGS_Float dsource; //source-isocentre distance
+       EGS_Float theta; //angle of rotation about Y-axis
+       EGS_Float phi;  //angle of rotation about Z-axis
+       EGS_Float phicol; //angle of rotation in source plane
+       EGS_Float mu; //monitor unit index for control point
+    };
+
+    /*! \brief Construct a dynamic source using \a Source as the
+    source
+    */
+    EGS_DynamicSource(EGS_BaseSource *Source,
+                          const string &Name="", EGS_ObjectFactory *f=0) :
+        EGS_BaseSource(Name,f), source(Source) {
+    };
+    /*! \brief Construct a dynamic source from the input \a inp */
```

Comment:
PS.  What's an astyle run and can I do it in the safety and comfort of my own home?